src/main/java/fish/payara/extras/diagnostics/collection/collectors/LogCollector.java

```diff
@@ -55,55 +55,79 @@
 public class LogCollector extends FileCollector {
 
     private Path logPath;
+    private Path accessLogPath;
+    private String logName;
+    private Boolean collectAccessLogs = false;
+    private Boolean collectNotificationLogs = false;
+    private Boolean collectServerLogs = false;
     private String dirSuffix;
 
-    public LogCollector(Path logPath) {
+    public LogCollector(Path logPath, String logName) {
         this.logPath = logPath;
+        this.logName = logName;
     }
 
-    public LogCollector(Path logPath, String instanceName) {
+    public LogCollector(Path logPath, String instanceName, String logName) {
         this.logPath = logPath;
         super.setInstanceName(instanceName);
+        this.logName = logName;
     }
 
-    public LogCollector(Path logPath, String instanceName, String dirSuffix) {
+    public LogCollector(Path logPath, String instanceName, String dirSuffix, String logName) {
         this.logPath = logPath;
         super.setInstanceName(instanceName);
         this.dirSuffix = dirSuffix;
+        this.logName = logName;
     }
 
+
+
     @Override
     public int collect() {
         Map<String, Object> params = getParams();
         if (params == null) {
             return 0;
         }
-        String outputPathString = (String) params.get((String) ParamConstants.DIR_PARAM);
+        String outputPathString = (String) params.get(ParamConstants.DIR_PARAM);
         Path outputPath = Paths.get(outputPathString, dirSuffix != null ? dirSuffix : "");
+
         if (confirmPath(logPath, false) && confirmPath(outputPath, true)) {
-            try {
-                logger.info("Collecting logs from " + (getInstanceName() != null ? getInstanceName() : "server"));
-                CopyDirectoryVisitor copyDirectoryVisitor = new CopyDirectoryVisitor(outputPath);
-                copyDirectoryVisitor.setInstanceName(getInstanceName());
-                Files.walkFileTree(logPath, copyDirectoryVisitor);
-            } catch (IOException io) {
-                logger.log(LogLevel.SEVERE, "Could not copy directory " + logPath.toString() + " to path " + outputPathString);
-                io.printStackTrace();
-                return 1;
-            }
+            collectLogs(logPath, outputPath.resolve("logs"), logName);
         }
+
         return 0;
     }
 
+    private void collectLogs(Path sourcePath, Path destinationPath, String fileContains) {
```

Comment:
If this fails, the collection will still report success, you should return an int